soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -4210,14 +4211,26 @@ s32 func_80839034(GlobalContext* globalCtx, Player* this, CollisionPoly* poly, u
                 func_800994A0(globalCtx);
             } else {
                 globalCtx->nextEntranceIndex = globalCtx->setupExitList[sp3C - 1];
+
+                // Main override for entrance rando and entrance skips
+                if (gSaveContext.n64ddFlag) {
+                    globalCtx->nextEntranceIndex = Entrance_OverrideNextIndex(globalCtx->nextEntranceIndex);
+                }
+
                 if (globalCtx->nextEntranceIndex == 0x7FFF) {
                     gSaveContext.respawnFlag = 2;
                     globalCtx->nextEntranceIndex = gSaveContext.respawn[RESPAWN_MODE_RETURN].entranceIndex;
                     globalCtx->fadeTransition = 3;
                     gSaveContext.nextTransition = 3;
                 } else if (globalCtx->nextEntranceIndex >= 0x7FF9) {
-                    globalCtx->nextEntranceIndex =
-                        D_808544F8[D_80854514[globalCtx->nextEntranceIndex - 0x7FF9] + globalCtx->curSpawn];
+                    // handle dynamic exits
+                    if (gSaveContext.n64ddFlag) {
```

Comment:
should we be checking for entrance rando here?

Reply:
Answered here: https://github.com/HarbourMasters/Shipwright/pull/1760#discussion_r1010976707